brew install pandoc github issue failure

### Dafny version

4.0.0

### Code to produce this issue

```dafny
Run brew install pandoc
```


### Command to run and resulting output

```code
brew install pandoc
  brew install --cask basictex
  eval "$(/usr/libexec/path_helper)"
  sudo tlmgr update --self
  sudo tlmgr install framed tcolorbox environ trimspaces unicode-math
  pandoc -v
  which latex || echo NOT FOUND latex
  which xelatex || echo NOT FOUND xelatex
  sudo gem install rouge -v 3.30.0
  shell: /bin/bash -e {0}
  env:
    DOTNET_ROOT: /Users/runner/.dotnet
==> Fetching pandoc
==> Downloading https://ghcr.io/v2/homebrew/core/pandoc/manifests/3.1
==> Downloading https://ghcr.io/v2/homebrew/core/pandoc/blobs/sha256:fff25a6f5ada0fd9b18859730b23cb3edfefc01cdc9a6483c726771932084f5d
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:fff25a6f5ada0fd9b18859730b23cb3edfefc01cdc9a6483c726771932084f5d?se=2023-03-20T18%3A10%3A00Z&sig=y7BOlMP7lb4Yh2eU0%2F5nKs1qXokdIh83x0gRLha24tA%3D&sp=r&spr=https&sr=b&sv=2019-12-12
==> Pouring pandoc--3.1.monterey.bottle.tar.gz
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
==> Summary
🍺  /usr/local/Cellar/pandoc/3.1: 10 files, 150.7MB
==> Running `brew cleanup pandoc`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Caveats
You must restart your terminal window for the installation of MacTex CLI tools to take effect.
Alternatively, Bash and Zsh users can run the command:

  eval "$(/usr/libexec/path_helper)"

==> Downloading https://mirror.ctan.org/systems/mac/mactex/mactex-basictex-20220314.pkg
==> Downloading from https://ctan.math.utah.edu/ctan/tex-archive/systems/mac/mactex/mactex-basictex-20220314.pkg
curl: (22) The requested URL returned error: 404
Error: Download failed on Cask 'basictex' with message: Download failed: https://mirror.ctan.org/systems/mac/mactex/mactex-basictex-20220314.pkg
Error: Process completed with exit code 1.
```


### What happened?

[brew install pandoc ](https://github.com/dafny-lang/dafny/blob/master/.github/workflows/refman.yml) github action tries to download https://mirror.ctan.org/systems/mac/mactex/mactex-basictex-20220314.pkg which throws a 404:

https://github.com/dafny-lang/dafny/actions/runs/4471353584/jobs/7856137115?pr=3762

Wrong version 2022.0314 (instead of 2023.0314) is hard-coded here: https://github.com/Homebrew/homebrew-cask/blob/master/Casks/basictex.rb

### What type of operating system are you experiencing the problem on?

Mac

--- a/.github/workflows/refman.yml
+++ b/.github/workflows/refman.yml
@@ -70,9 +70,8 @@ jobs:
     - name: Install latex pandoc - MacOS
       if: runner.os == 'MacOS'
       run: |
-        brew update-reset && brew update
         brew install pandoc
-        brew install --cask mactex-no-gui
+        brew install --cask mactex
         eval "$(/usr/libexec/path_helper)"
         sudo tlmgr update --self
         sudo tlmgr install framed tcolorbox environ trimspaces unicode-math

--- a/.github/workflows/refman.yml
+++ b/.github/workflows/refman.yml
@@ -70,8 +70,9 @@ jobs:
     - name: Install latex pandoc - MacOS
       if: runner.os == 'MacOS'
       run: |
-        brew install --cask basictex
+        brew update-reset && brew update
         brew install pandoc
+        brew install --cask basictex
         eval "$(/usr/libexec/path_helper)"
         sudo tlmgr update --self
         sudo tlmgr install framed tcolorbox environ trimspaces unicode-math

--- a/.github/workflows/refman.yml
+++ b/.github/workflows/refman.yml
@@ -71,6 +71,7 @@ jobs:
       if: runner.os == 'MacOS'
       run: |
         brew update
+        brew upgrade
         brew install pandoc
         brew install --cask basictex
         eval "$(/usr/libexec/path_helper)"

--- a/.github/workflows/refman.yml
+++ b/.github/workflows/refman.yml
@@ -72,7 +72,7 @@ jobs:
       run: |
         brew update-reset && brew update
         brew install pandoc
-        brew cask install mactex-no-gui
+        brew install --cask mactex-no-gui
         eval "$(/usr/libexec/path_helper)"
         sudo tlmgr update --self
         sudo tlmgr install framed tcolorbox environ trimspaces unicode-math

--- a/.github/workflows/refman.yml
+++ b/.github/workflows/refman.yml
@@ -72,7 +72,7 @@ jobs:
       run: |
         brew update-reset && brew update
         brew install pandoc
-        brew install --cask mactex
+        brew install --cask basictex
         eval "$(/usr/libexec/path_helper)"
         sudo tlmgr update --self
         sudo tlmgr install framed tcolorbox environ trimspaces unicode-math

--- a/.github/workflows/refman.yml
+++ b/.github/workflows/refman.yml
@@ -70,10 +70,8 @@ jobs:
     - name: Install latex pandoc - MacOS
       if: runner.os == 'MacOS'
       run: |
-        brew update
-        brew upgrade
-        brew install pandoc
         brew install --cask basictex
+        brew install pandoc
         eval "$(/usr/libexec/path_helper)"
         sudo tlmgr update --self
         sudo tlmgr install framed tcolorbox environ trimspaces unicode-math

--- a/.github/workflows/refman.yml
+++ b/.github/workflows/refman.yml
@@ -70,6 +70,7 @@ jobs:
     - name: Install latex pandoc - MacOS
       if: runner.os == 'MacOS'
       run: |
+        brew update
         brew install pandoc
         brew install --cask basictex
         eval "$(/usr/libexec/path_helper)"

--- a/.github/workflows/refman.yml
+++ b/.github/workflows/refman.yml
@@ -72,7 +72,7 @@ jobs:
       run: |
         brew update-reset && brew update
         brew install pandoc
-        brew install --cask basictex
+        brew cask install mactex-no-gui
         eval "$(/usr/libexec/path_helper)"
         sudo tlmgr update --self
         sudo tlmgr install framed tcolorbox environ trimspaces unicode-math

--- a/.github/workflows/refman.yml
+++ b/.github/workflows/refman.yml
@@ -70,6 +70,7 @@ jobs:
     - name: Install latex pandoc - MacOS
       if: runner.os == 'MacOS'
       run: |
+        brew update-reset && brew update
         brew install pandoc
         brew install --cask mactex
         eval "$(/usr/libexec/path_helper)"

